File: app/src/main/java/com/solution/it/newsoft/dagger/AppComponent.java
package com.solution.it.newsoft.dagger;

import com.solution.it.newsoft.NewSoftApp;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        ActivityModules.class,
        ListingModules.class
})
public interface AppComponent extends AndroidInjector<NewSoftApp> {
}

File: app/src/main/java/com/tengyeekong/kotlinplayground/dagger/AppComponent.kt
package com.tengyeekong.kotlinplayground.dagger

import android.app.Application

import javax.inject.Singleton

import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.DaggerApplication
import dagger.android.support.AndroidSupportInjectionModule

@Singleton
@Component(modules = [AndroidSupportInjectionModule::class, ActivityBindingModules::class, AppModules::class])
interface AppComponent : AndroidInjector<DaggerApplication> {

    override fun inject(instance: DaggerApplication)

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent
    }
}
